clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/model/InstanceProvider.groovy

```diff
@@ -26,5 +26,5 @@ interface InstanceProvider<T extends Instance> {
 
   T getInstance(String account, String region, String id)
 
-  String getConsoleOutput(String account, String region, String id)
+  S getConsoleOutput(String account, String region, String id)
```

Comment:
I'd prefer if this interface had two different methods: `getConsoleOutput` and `getConsoleOutputs`. It would implement a `default` for `getConsoleOutputs` that would return an empty list.

The return type of `InstanceController.getConsoleOutput` would be a map with both `output` and `outputs`, which would be populated by `getConsoleOutput` and `getConsoleOutputs`, respectively.

That way, the return type of `InstanceController.getConsoleOutput` will be consistent, and there won't be any special-casing in the UI for Kubernetes. If there are `outputs`, render those, otherwise, render `output`.

Reply:
From my perspective I like this current approach -- the caller should know what to expect, and we don't litter the interfaces with more fields.